### PR TITLE
Enable embedded config for aarch64

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow to run AArch64 version of clickhouse-server without configs. This facilitates #15174

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Waiting only builds.

**Comment by alexey-milovidov** [comment]
Yes, it started to work.

File: programs/server/Server.cpp [hunk 1]
```diff
@@ -339,16 +339,23 @@ int Server::main(const std::vector<std::string> & /*args*/)
...
-                /// It's better to lock only the code segment instead of calling "mlockall",
-                /// because otherwise debug info will be also locked in memory, and it can be huge.
-                auto [addr, len] = getMappedArea(reinterpret_cast<void *>(mainEntryClickHouseServer));
-
-                LOG_TRACE(log, "Will do mlock to prevent executable memory from being paged out. It may take a few seconds.");
-                if (0 != mlock(addr, len))
-                    LOG_WARNING(log, "Failed mlock: {}", errnoToString(ErrorCodes::SYSTEM_ERROR));
-                else
-                    LOG_TRACE(log, "The memory map of clickhouse executable has been mlock'ed, total {}", ReadableSize(len));
+                try
```
Comment (filimonov) [inline programs/server/Server.cpp:342]: That should prevent situations like #14400

File: programs/server/CMakeLists.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ set(CLICKHOUSE_SERVER_SOURCES
     Server.cpp
 )
 
-if (OS_LINUX AND ARCH_AMD64)
```
Comment (filimonov) [inline programs/server/CMakeLists.txt:6]: BTW - that issue didn't popup on your tests https://github.com/ClickHouse/ClickHouse/issues/14002 ?

Comment (alexey-milovidov) [reply programs/server/CMakeLists.txt:6]: Official build does not have that issue as we use `llvm-objcopy` instead of `objcopy`.

**Pull request merged**